orderbook/src/api/cancel_order.rs

```diff
@@ -1 +1,75 @@
-// TODO - write me.
\ No newline at end of file
+use crate::orderbook::{Orderbook, OrderCancellationResult};
+use anyhow::Result;
+use model::order::OrderCancellation;
+use std::{convert::Infallible, sync::Arc};
+use warp::{hyper::StatusCode, Filter, Rejection, Reply};
+
+const MAX_JSON_BODY_PAYLOAD: u64 = 1024 * 16;
+
+fn extract_cancellation() -> impl Filter<Extract = (OrderCancellation,), Error = Rejection> + Clone {
```

Comment:
This can probably be shared with `extract_user_order` and become a more abstract `fn extract_payload<T: DeserializeOwned + Send>() -> impl Filter<Extract = (T,), Error = Rejection> + Clone { ... }
`